test/metabase/util/debugging_test.clj

```diff
@@ -0,0 +1,43 @@
+(ns metabase.util.debugging-test
+  (:require  [clojure.test :refer :all]
+             [metabase.config :as config]))
+
+(def dbg-prn-used? (atom false))
+
+(defn dbg-prn*
+  "Implementation for the `dbg-prn` macro."
+  [p x]
+  (p x)
+  x)
+
+(defmacro dbg-prn
+  "`prn`s the argument and also returns it, which is useful for debugging. There is also a CI check to ensure no usages
+  of it sneak into master. With two arguments, uses the first as a printing function instead of `prn`."
+  ([x]
+   `(dbg-prn prn ~x))
+  ([p x]
+   `(do
+      (when-not (= (ns-name *ns*) 'metabase.util.debugging-test)
+        (reset! metabase.util.debugging-test/dbg-prn-used? true))
+      (dbg-prn* ~p ~x))))
+
+(deftest test-dbg-prn-is-not-used
+  (testing "`dbg-prn` should not be committed to master"
+    (is (not @dbg-prn-used?))))
```

Comment:
Everything about writing this test gave me heartburn, but I can confirm that it failed when I put a `dbg-prn` elsewhere in the codebase (with a dummy commit). I especially don't like having a separate ns and comingling the implementation and the testing, but I couldn't think of a nicer way to do it.

I feel like this would be a simple feature to add to Eastwood, but that'd be a whole 'nother project.